whitelist.py

```diff
@@ -340,6 +345,30 @@ def get_contents(self) -> dict:
 
         return self.__load_contents()
 
+    def mark_status(self, user_id: int, status: bool = False) -> None:
+        """
+        Marks a user as offline or online
+
+        Args:
+            user_id (int): The user to mark as online or offline
+            status (bool): True for online, False for offline
+
+        Returns:
+            None
+
+        Raises:
+            ValueError: If the user does not exist
+        """
+        if not self.exists(user_id):
+            raise ValueError(f'Cannot flag user "{user_id}" as offline as they do not exist.')
+
+        contents = self.__load_contents()
```

Comment:
I'm loading the contents of the json file in many methods in this class, sometimes I load the contents then reference a method which in turn loads the contents to do its checks.

Consider moving the contents into a class property and only updating it when the file is saved. To be handled in a future branch / PR.